teps/0052-tekton-results-automated-run-resource-cleanup.md

```diff
@@ -211,3 +329,26 @@ Watcher for a few reasons:
   and knows whether a Run has been successfully recorded. It makes sense to hook
   into this lifecycle to minimize resource footprint and reuse logic that's
   already present.
+
+### Be smarter about how Tekton Controllers query resources
+
+If the main source of issues are the List calls being made, we could potentially
+look into modifying how Tekton controllers look up existing data and attempt to
+remove the latency introduced by List calls (exactly how we could do this TBD).
```

Comment:
i mentioned this in a comment above but i think we should do this regardless of this proposal (if it is actually a problem)

Reply:
Data I pulled so far suggests this isn't that big of an issue for controllers since most of the data is cached. Not sure if this would differ with real CRD types (e.g. would we start getting memory constrained?) We can follow up on this in the other issue.